ext/ldap/ldap.c

```diff
@@ -1016,6 +1016,77 @@ PHP_FUNCTION(ldap_connect)
 }
 /* }}} */
 
+#ifdef HAVE_ORALDAP
+#ifdef LDAP_API_FEATURE_X_OPENLDAP
+PHP_FUNCTION(ldap_connect_wallet) {
+	char *host = NULL;
+	size_t hostlen = 0;
+	char *wallet = NULL, *walletpasswd = NULL;
+	size_t walletlen = 0, walletpasswdlen = 0;
+	zend_long authmode = GSLC_SSL_NO_AUTH;
+	int ssl = 0;
+
+	ldap_linkdata *ld;
+	LDAP *ldap = NULL;
+
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "s!ss|l",
+		&host, &hostlen, &wallet, &walletlen, &walletpasswd, &walletpasswdlen, &authmode) != SUCCESS
+	) {
+		RETURN_THROWS();
+	}
+
+	if (authmode != 0) {
+		ssl = 1;
+	}
+
+	if (LDAPG(max_links) != -1 && LDAPG(num_links) >= LDAPG(max_links)) {
+		php_error_docref(NULL, E_WARNING, "Too many open links (" ZEND_LONG_FMT ")", LDAPG(num_links));
+		RETURN_FALSE;
+	}
```

Comment:
Kinda hate that we mimic the API by returning false on failure....

Reply:
I'm fine with throwing an exception here. I guess we have this much flexibility with the implementation. :)

Reply:
ON the flip side, doing so is not consistent with the behavior of `ldap_connect`. :S So I'm unsure overall

Reply:
I would stay consistent. Otherwise the migration path is not straight forward.